### ISSUE TITLE
[explicit-member-accessibility] Throws error even though begins with #

### ISSUE THREAD

**Issue by tscpp** [typescript-eslint/typescript-eslint#2175]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": "error"
  }
}
```

```TS
class MyClass {
  #myPrivateProperty = 'super private';
}
```

**Expected Result**
The plugin to not throw an error when the property is marked private with `#`.

**Actual Result**
The plugin throws an error even though marked as private with `#`.

**Additional Info**
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.2` |
| `@typescript-eslint/parser`        | `3.0.2` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.3.0` |
| `npm`                              | `6.14.4` |

**Comment by tscpp** [comment]
Duplicate [#1436](https://github.com/typescript-eslint/typescript-eslint/issues/1436)
